pdns/dnsdistdist/dnsdist-lua-bindings-dnsquestion.cc

```diff
@@ -29,6 +29,28 @@
 #include "dnsdist-snmp.hh"
 #include "dnsparser.hh"
 
+#include "protozero.hh"
+
+static void addMetaKeyAndValuesToProtobufContent(DNSQuestion& dnsQuestion, const std::string& key, const LuaArray<boost::variant<int64_t, std::string>>& values)
+{
+#if !defined(DISABLE_PROTOBUF)
+  protozero::pbf_writer pbfWriter{dnsQuestion.d_rawProtobufContent};
+  protozero::pbf_writer pbfMetaWriter{pbfWriter, static_cast<protozero::pbf_tag_type>(pdns::ProtoZero::Message::Field::meta)};
+  pbfMetaWriter.add_string(static_cast<protozero::pbf_tag_type>(pdns::ProtoZero::Message::MetaField::key), key);
+  protozero::pbf_writer pbfMetaValueWriter{pbfMetaWriter, static_cast<protozero::pbf_tag_type>(pdns::ProtoZero::Message::MetaField::value)};
+  for (const auto& value : values) {
+    if (value.second.type() == typeid(std::string)) {
+      pbfMetaValueWriter.add_string(static_cast<protozero::pbf_tag_type>(pdns::ProtoZero::Message::MetaValueField::stringVal), boost::get<std::string>(value.second));
+    }
+    else {
+      pbfMetaValueWriter.add_uint64(static_cast<protozero::pbf_tag_type>(pdns::ProtoZero::Message::MetaValueField::intVal), boost::get<int64_t>(value.second));
+    }
+  }
+  pbfMetaValueWriter.commit();
+  pbfMetaWriter.commit();
+#endif /* DISABLE_PROTOBUF */
```

Comment:
Maybe, for consistency?
```suggestion
  pbfWriter.commit();
#endif /* DISABLE_PROTOBUF */
```

Reply:
I'll test to be sure but IIRC we cannot call `commit()` on the top-level writer.

Reply:
```
testrunner: ../ext/protozero/include/protozero/basic_pbf_writer.hpp:356: void protozero::basic_pbf_writer<TBuffer>::commit() [with TBuffer = std::__cxx11::basic_string<char>]: Assertion `m_parent_writer && "you can't call commit() on a basic_pbf_writer without a parent"' failed.
unknown location(0): fatal error: in "test_dnsdist_lua_ffi/test_meta_values": signal: SIGABRT (application abort requested)
../test-dnsdist-lua-ffi.cc(1078): last checkpoint
```
Indeed we cannot :)